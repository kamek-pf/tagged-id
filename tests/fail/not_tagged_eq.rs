use tagged_id::{Id, Identify};

struct TestThis();
struct TestThat();

impl Identify for TestThis {
    type InnerId = i32;
}

impl Identify for TestThat {
    type InnerId = i32;
}

fn main() {
    let id1: Id<TestThis> = 42.into();
    let id2: Id<TestThat> = 42.into();

    assert_eq!(id1, id2);
}
