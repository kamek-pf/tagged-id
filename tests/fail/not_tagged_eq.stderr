error[E0308]: mismatched types
  --> tests/fail/not_tagged_eq.rs:18:21
   |
18 |     assert_eq!(id1, id2);
   |                     ^^^ expected `Id<TestThis>`, found `Id<TestThat>`
   |
   = note: expected struct `Id<TestThis>`
              found struct `Id<TestThat>`
